<?xml version="1.0"?>
<launch>
    <!-- Must uniquely identify the microphone  -->
    <arg name="partial_name" default="Lavalier" />

    <!-- In Hz -->
    <arg name="sample_freq" default="44100" />

    <!-- Length, in s, of publish buffer -->
    <arg name="buffer_time" default="0.25" /> 

    <!-- How often to transfer buffered data from input stream to publishing buffer -->
    <arg name="block_time" default="0.005" />

    <!-- How often to publish most recent buffer -->
    <arg name="publish_freq" default="60" /> 

    <arg name="node_name" default="mic_streamer" />

    <node pkg="multicam_server" type="mic_streamer.py" name="$(arg node_name)" output="screen" >
        <param name="publish_freq" type="double" value = "$(arg publish_freq)" /> 
        <param name="partial_name" type="string" value="$(arg partial_name)" /> 
        <param name="sample_freq" type="int" value="$(arg sample_freq)" /> 
        <param name="block_time" type="double" value="$(arg block_time)" />
        <param name="buffer_time" type="double" value="$(arg buffer_time)" /> 
    </node>
</launch>
